samples/drivers/watchdog/sample.yaml

```diff
@@ -27,3 +27,15 @@ tests:
     platform_allow: b_u585i_iot02a nucleo_f091rc nucleo_f103rb nucleo_f207zg nucleo_f429zi
         nucleo_f746zg nucleo_g071rb nucleo_g474re nucleo_h743zi nucleo_l073rz nucleo_l152re
         nucleo_wb55rg nucleo_wl55jc stm32f3_disco stm32l562e_dk disco_l475_iot1
+  sample.drivers.watchdog.gd32_fwdgt:
+    extra_args: DTC_OVERLAY_FILE=boards/gd32_fwdgt.overlay
+    filter: dt_compat_enabled("gd,gd32-fwdgt")
+    platform_allow: gd32e103v_eval gd32e507v_start gd32f350r_eval gd32f403z_eval
+        gd32f450i_eval gd32f450z_eval gd32f470i_eval gd32vf103c_starter
+        gd32vf103v_eval longan_nano
+  sample.drivers.watchdog.gd32_wwdgt:
+    extra_args: DTC_OVERLAY_FILE=boards/gd32_wwdgt.overlay
+    filter: dt_compat_enabled("gd,gd32-wwdgt")
+    platform_allow: gd32e103v_eval gd32e507v_start gd32f350r_eval gd32f403z_eval
+        gd32f450i_eval gd32f450z_eval gd32f470i_eval gd32vf103c_starter
+        gd32vf103v_eval longan_nano
```

Comment:
this seems redundant with the filter, same above.

Reply:
Rewrite with filter: property

Reply:
This test gives the DTC_OVERLAY_FILE option. It causes compile failure on other platforms.
This `platform_allow` line is needed.